File: problems/Best Time to Buy and Sell Stock II/maxProfit.cs
public class Solution {
    public int MaxProfit(int[] prices) {
        var max = 0;

        for (var i = 1; prices.Length > i; ++i) {
            if (prices[i] > prices[i - 1]) {
                max += prices[i] - prices[i - 1];
            }
        }

        return max;
    }
}

File: problems/Best Time to Buy and Sell Stock with Transaction Fee/maxProfit.java
class Solution {
    public int maxProfit(int[] prices, int fee) {
        int cash = 0;
        int hold = -prices[0];

        for (int price : prices) {
            cash = Math.max(cash, hold + price - fee);
            hold = Math.max(hold, cash - price);
        }

        return cash;
    }
}
